File: 绿盟全国赛决赛参考答案/Java/第一阶段/step3/Task.java
package step3;

public class Task {
	
	public int concaveValley(Integer[] arr){
		/********* Begin *********/
		int res = 0;
		int up = 0;
		int down = 0;
		for (int i = 1; i < arr.length; i++) {
			if(up > 0 && arr[i - 1] > arr[i] || arr[i - 1] == arr[i])  up = down = 0;					
			if(arr[i - 1] < arr[i])   up +=1;
			if(arr[i - 1] > arr[i])   down+=1;
			if( up > 0 && down > 0 && up + down + 1 > res)   res = up + down + 1;		
			
		}
		return res;
		/********* End *********/	
	}
}

File: 绿盟全国赛决赛参考答案/Python/第一阶段/step3/Task.py
# -*- coding: utf-8 -*-
class Task:

    def concaveValley(self,X):
        res = 0
        up = 0
        down = 0
        for i in range(1, len(X)):
            if up > 0 and X[i - 1] > X[i] or X[i - 1] == X[i]:
                up = down = 0
            if X[i - 1] < X[i]:
                up += 1
            if X[i - 1] > X[i]:
                down += 1
            if up > 0 and down > 0 and up + down + 1 > res:
                res = up + down + 1
        return res
